file/1.3/modules/ROOT/pages/file-documentation.adoc

```diff
@@ -663,11 +663,22 @@ A matcher can also be used for additional filtering of files.
 [[time-between-size-check]]
 === Time Between Size Check Parameter Behavior
 
-This wait is performed at the moment of retrieving the content of the file.
-If the flow never accesses the content of the file, the check is not done. Also,
-because this wait can take place in multiple message processors, this connector has no
-control over which thread will be the one performing the wait, thus leaving the possibility
-open to block threads not meant for this type of behavior.
+The process of the Write operation on a file can take some time to complete and get the written-to file ready to be read or accessed.  Thereby, MuleSoft introduced the `timeBetweenSizeCheck` parameter to prevent an operation from accessing an incomplete file.
+When you set this parameter, it verifies the file's size before accessing the file contents.
+The verification validates that the file's size does not vary after
+waiting for the amount of time specified by the parameters `timeBetweenSizeCheck` and `timeBetweenSizeCheckUnit`.
+The `timeBetweenSizeCheck` parameter then indicates the time to wait between each
+verification, which is performed twice. If the file's size remains the same during both times, this assumes that the Write operation is complete and the access operation can continue, otherwise an exception is thrown.
+The performed wait indicates that the thread executing
+the operation goes to sleep for the indicated period of time. This wait is realized
+at the moment of retrieving the content of the file. If the flow never accesses
+the content of the file, the check is not done.
+
+[NOTE]
+In Mule runtime engine version 4.1 and 4.2 this wait can take place in multiple message processors and the File Connector
```

Comment:
Add a comma after "4.2."
"...over which thread performs the wait."

Again, for brevity and trying not to use passive voice:  "Any thread from any thread pool, even a CPU-intensive thread, can perform the wait.

Reply:
Also, CPU is usually all caps.

Reply:
"such wait" > "the wait."

Reply:
Does it make sense to say "... wait process" instead of "wait"?